### PR TITLE
feat(sh-admin): enhanced user management in admin dashboard

### PR DESCRIPTION
**Pull request by JoelJacobStephen**
### Ticket
- Closes HFE-406
- Depends on #3786 

### Description
This PR brings a ton of features to improve user management in Admin Dashboard. 

### New Features
1. Admins can now update the name of any user,
2. Bulk actions are now possible in the users table. These actions include:
	- Elevation of multiple users to admin status,
	- Removal of admin status from multiple users,
	- Deletion of multiple users,
3. Ability to search through the users table by name or email address of the users.
4. New page based pagination system for the users table replacing the previous used scroll based pagination system.
5. Ability to revoke pending invites from the pending invites page.

### Additional Changes
Due to the changes introduced in `HoppSmartTable`, all tables in the admin dashboard have been updated to adhere to the new changes.

### Screenshot
image

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Review by jamesgeorge007 — CHANGES_REQUESTED** [review]
Search and pagination don't play well in the user list view. The computation for total pages doesn't account for search in all cases.

**Comment by JoelJacobStephen** [comment]
> Search and pagination don't play well in the user list view. The computation for total pages doesn't account for search in all cases.

@jamesgeorge007 I've fixed the issue. You can test it now.

**Review by jamesgeorge007 — CHANGES_REQUESTED** [review]
- A search session has to ensure the page number gets reset to `1`.
- At times the entire list gets populated while clearing the search entry at once.

**Comment by JoelJacobStephen** [comment]
> * A search session has to ensure the page number gets reset to `1`.
> * At times the entire list gets populated while clearing the search entry at once.

@jamesgeorge007 I have resolved these issue.

File: packages/hoppscotch-sh-admin/src/pages/users/index.vue
Comment (jamesgeorge007) [inline packages/hoppscotch-sh-admin/src/pages/users/index.vue:1]: There's an issue arising here while changing the admin status (making a user admin & vice versa) for a user and immediately selecting the row for which the action was performed won't result in the action bar (with toggle admin & delete actions) to show up. It shows up only during the second attempt.

File: packages/hoppscotch-sh-admin/src/pages/users/index.vue [hunk 1]
```diff
@@ -15,132 +15,182 @@
...
+            :icon="IconRight"
+            :disabled="page >= totalPages"
+            @click="changePage(PageDirection.Next)"
+          />
+        </div>
+
+        <HoppSmartTable
+          :headings="headings"
+          :list="finalUsersList"
+          :loading="showSpinner"
```
Comment (jamesgeorge007) [inline packages/hoppscotch-sh-admin/src/pages/users/index.vue:49]: There's a slight difference in height compared to the spinner and the row with actual data. We could leverage the `loading-state` slot to implement a custom spinner to fine-tune the dimensions.

**Review by jamesgeorge007 — APPROVED** [review]
LGTM 🚀 
Leaving a few pointers below for future reference.

File: packages/hoppscotch-sh-admin/src/pages/users/index.vue
Comment (jamesgeorge007) [reply packages/hoppscotch-sh-admin/src/pages/users/index.vue:1]: This requires an update to [@hoppscotch/ui](https://github.com/hoppscotch/ui) and will be handled separately.

**Review by nivedin — CHANGES_REQUESTED** [review]

File: packages/hoppscotch-sh-admin/src/pages/teams/index.vue [hunk 2]
```diff
@@ -18,89 +18,83 @@
 
         <div v-else-if="error">{{ t('teams.load_list_error') }}</div>
 
-        <HoppSmartTable v-else-if="teamsList.length" :list="teamsList">
+        <HoppSmartTable
+          v-else-if="teamsList.length"
+          :headings="headings"
+          :list="teamsList"
+          @onRowClicked="goToTeamDetails"
```
[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply packages/hoppscotch-sh-admin/src/pages/teams/index.vue:25]: @JoelJacobStephen, please make a note of these to address separately.

**Pull request merged**